ota_university/ctf_framework/views/challenge.py

```diff
@@ -0,0 +1,70 @@
+from django.contrib.auth.decorators import login_required
+from .base_view import *
+
+
+@login_required()
+def index(request):
+    user = UserProfile.objects.get(user=request.user)
+    challenges = Challenge.objects.all()
+
+    challenge_context = []
+
+    for challenge in challenges:
+        challenge_context.append({"challenge": challenge, "completed": challenge in user.challenges.all()})
+
+    context = {
+        "challenges": challenge_context
+    }
+    return render(request, "challenge/index.html", context)
+
+
+@login_required()
+def show(request, challenge_id):
+    """View the page of a specific challenge."""
+    user = UserProfile.objects.get(user=request.user)
+    message = None
+
+    challenge = Challenge.objects.get(id=challenge_id)
+
+    context = {
+        "challenge": challenge,
+        "user": user,
+        "challenge_completed": challenge in user.challenges.all(),
+        "message": message
```

Comment:
Is message used here? I see it being set to `None` and isn't used elsewhere.

Reply:
An artifact of the view refactor. Removing.